module-apps/application-phonebook/windows/PhonebookSearchResults.cpp

```diff
@@ -65,25 +82,31 @@ namespace gui
             return false;
         }
 
-        auto searchResultsData = dynamic_cast<PhonebookSearchResultsData *>(data);
+        const auto searchResultsData = dynamic_cast<PhonebookSearchResultsData *>(data);
         assert(searchResultsData != nullptr);
 
-        searchResultsModel = searchResultsData->consumeSearchResultsModel();
-        setTitle(utils::translate("common_results_prefix") + "\"" + searchResultsModel->getFilter() + "\"");
+        searchResultsModel = searchResultsData->model;
         searchResultList->setProvider(searchResultsModel);
+        handleContentUpdate();
 
-        if (searchResultsModel->contactSelectCallback) {
-            navBar->setActive(nav_bar::Side::Left, false);
-            navBar->setText(nav_bar::Side::Left, "");
-            navBar->setText(nav_bar::Side::Center, utils::translate(style::strings::common::select));
-        }
-        else {
+        return true;
+    }
+    void PhonebookSearchResults::handleContentUpdate()
+    {
+        setTitle(utils::translate("common_results_prefix") + "\"" + searchResultsModel->getFilter() + "\"");
+
+        if (const auto resultCount = searchResultsModel->requestRecordsCount()) {
```

Comment:
As far as I can see `resultCount` isn't used anywhere, why not just `searchResultsModel->requestRecordsCount() > 0`?

Reply:
You're right, I must have missed it when I copy-pasted this piece of code :)